homeassistant/components/ted/config_flow.py

```diff
@@ -0,0 +1,134 @@
+"""Config flow for TED integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import httpx
+import tedpy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .const import (
+    CONF_MTU_ENERGY_DAILY,
+    CONF_MTU_ENERGY_MTD,
+    CONF_MTU_ENERGY_NOW,
+    CONF_MTU_POWER_VOLTAGE,
+    CONF_SPYDER_ENERGY_DAILY,
+    CONF_SPYDER_ENERGY_MTD,
+    CONF_SPYDER_ENERGY_NOW,
+    DOMAIN,
+    OPTION_DEFAULTS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TED."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize an TED flow."""
+        self.name = None
+        self.serial = None
+        self.model = None
+
+    @callback
+    def _async_generate_schema(self):
+        """Generate schema."""
+        schema = {vol.Required(CONF_HOST): str}
+
+        return vol.Schema(schema)
+
+    @callback
+    def _async_current_hosts(self):
+        """Return a set of hosts."""
+        return {
+            entry.data[CONF_HOST]
+            for entry in self._async_current_entries(include_ignore=False)
+            if CONF_HOST in entry.data
+        }
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            if user_input[CONF_HOST] in self._async_current_hosts():
+                return self.async_abort(reason="already_configured")
```

Comment:
Could this be done using `_abort_if_unique_id_configured()` rather than a custom function?